cwms-data-api/src/main/java/cwms/cda/data/dto/location/kind/CharacteristicRef.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (c) 2024. Hydrologic Engineering Center (HEC).
+ * United States Army Corps of Engineers
+ * All Rights Reserved. HEC PROPRIETARY/CONFIDENTIAL.
+ * Source may not be released without written approval from HEC
+ */
+
+package cwms.cda.data.dto.location.kind;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.databind.PropertyNamingStrategies;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonNaming;
+import cwms.cda.api.errors.FieldException;
+import cwms.cda.api.errors.RequiredFieldException;
+import cwms.cda.data.dto.CwmsDTOBase;
+import cwms.cda.formatters.Formats;
+import cwms.cda.formatters.annotations.FormattableWith;
+import cwms.cda.formatters.json.JsonV2;
+import cwms.cda.formatters.xml.XMLv2;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+@FormattableWith(contentType = Formats.JSONV2, formatter = JsonV2.class, aliases = {Formats.DEFAULT, Formats.JSON})
+@FormattableWith(contentType = Formats.XMLV2, formatter = XMLv2.class, aliases = {Formats.XML})
+@JsonDeserialize(builder = CharacteristicRef.Builder.class)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonNaming(PropertyNamingStrategies.KebabCaseStrategy.class)
+public final class CharacteristicRef implements CwmsDTOBase
```

Comment:
This could totally be a CwmsId, after looking at it.  I think that's the right way to go.
Also, as for the "Ref" part of the class, it might be a bit of a misnomer.  I named it after the db type it's derived from, which is CHARACTERISTIC_REF_T.

Reply:
Replaced with CwmsId